system-addon/lib/SectionsManager.jsm

```diff
@@ -278,11 +278,16 @@ class SectionsFeed {
       case at.PREFS_INITIAL_VALUES:
         SectionsManager.init(action.data);
         break;
-      case at.PREF_CHANGED:
-        if (action.data && action.data.name.match(/^feeds.section.(\S+).options$/i)) {
-          SectionsManager.addBuiltInSection(action.data.name.slice(0, -8), action.data.value);
+      case at.PREF_CHANGED: {
+        if (action.data) {
+          const matched = action.data.name.match(/^feeds.section.(\S+).options$/i);
+          if (matched) {
+            SectionsManager.addBuiltInSection(action.data.name.slice(0, -8), action.data.value);
+            this.store.dispatch({type: at.SECTION_OPTIONS_CHANGED, data: matched[1]});
```

Comment:
I'm not super familiar with this code, but is this the way to talk to sections? E.g., not a direct method or `emit` or something?

Reply:
Yes, I think so. ```emit``` it uses to relay events to itself.